### ISSUE TITLE
learning rate schedulers

### ISSUE THREAD

**Issue by Plyb** [DRAGNLabs/MIRROR-Pipeline#192]
**Acceptance Criteria**:
- `fit` accepts a customizable learning rate scheduler
- an example for specifying the lr scheduler is present

**Possible design**:
- `Trainer.fit` accepts a `configure_scheduler: ConfigureScheduler | None` parameter
  - If present, a scheduler is constructed using the optimizer
  - if a scheduler is constructed, `scheduler.step()` is called after `optimizer.step()`
  - `class ConfigureScheduler(Protocol): def __call__(self, optimizer: Optimizer, total_training_steps: int) -> Scheduler: ...`
    - `Optimizer` and `Scheduler` are both the built in torch types.
    - You'll need to implement an example of ConfigureScheduler
- `configure_optimizer` flows from the subcommand `fit`'s parameters.

**Comment by morrisoncheney** [comment]
<img width="537" height="307" alt="Image" src="https://github.com/user-attachments/assets/ff27813e-2679-44c5-8e9f-1569fe9deedc" />

<img width="538" height="305" alt="Image" src="https://github.com/user-attachments/assets/7b5ec931-141e-4e70-84ac-b4b74c9ea989" />

The first is using cosine annealing. The second is not. Both had seed (which is implemented in the llama model to be used with random initialization) set to 42.
